File: src/main/java/cn/solarmoon/idyllic_food_diary/element/matter/cookware/container/plate/PlateBlockRenderer.java
package cn.solarmoon.idyllic_food_diary.element.matter.cookware.container.plate;

import cn.solarmoon.idyllic_food_diary.element.matter.cookware.container.ContainerBlockEntity;
import cn.solarmoon.idyllic_food_diary.feature.basic_feature.RendererUtil;
import cn.solarmoon.solarmoon_core.api.renderer.BaseBlockEntityRenderer;
import com.mojang.blaze3d.vertex.PoseStack;
import net.minecraft.client.renderer.MultiBufferSource;
import net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider;

public class PlateBlockRenderer extends BaseBlockEntityRenderer<ContainerBlockEntity> {

    public PlateBlockRenderer(BlockEntityRendererProvider.Context context) {
        super(context);
    }

    @Override
    public void render(ContainerBlockEntity plate, float v, PoseStack poseStack, MultiBufferSource buffer, int light, int overlay) {
        RendererUtil.renderItemStackStack(0.5f, 45, 2, 0.5f,
                plate, poseStack, buffer, light, overlay, context);
    }

    @Override
    public boolean shouldRenderOffScreen(ContainerBlockEntity p_112306_) {
        return true;
    }

}

File: src/main/kotlin/cn/solarmoon/idyllic_food_diary/element/matter/food_container/plate/PlateBlockRenderer.kt
package cn.solarmoon.idyllic_food_diary.element.matter.food_container.plate

import cn.solarmoon.idyllic_food_diary.element.matter.food_container.long_plate.LongPlateBlockEntity
import cn.solarmoon.idyllic_food_diary.feature.util.RenderUtil
import cn.solarmoon.spark_core.api.renderer.HandyBlockEntityRenderer
import com.mojang.blaze3d.vertex.PoseStack
import net.minecraft.client.Minecraft
import net.minecraft.client.renderer.MultiBufferSource
import net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider
import net.minecraft.world.phys.Vec3

class PlateBlockRenderer(context: BlockEntityRendererProvider.Context): HandyBlockEntityRenderer<PlateBlockEntity>(context) {

    override fun render(
        blockEntity: PlateBlockEntity,
        partialTick: Float,
        poseStack: PoseStack,
        bufferSource: MultiBufferSource,
        packedLight: Int,
        packedOverlay: Int
    ) {
        RenderUtil.renderItemStackStack(0.5f, 45f, 2f, 0.5f, blockEntity, poseStack, bufferSource, packedLight, packedOverlay, context)
    }

    override fun shouldRender(blockEntity: PlateBlockEntity, cameraPos: Vec3): Boolean {
        return true
    }

    override fun shouldRenderOffScreen(blockEntity: PlateBlockEntity): Boolean {
        return true
    }

}